util/bklog/log.go

```diff
@@ -12,6 +13,10 @@ var (
 	L = logrus.NewEntry(logrus.StandardLogger())
 )
 
+var (
+	spanExporter, _ = detect.Exporter()
```

Comment:
we shouldn't call `detect` in here as initialization based on env should only come from main packages. If we can't determine if the context is valid tracer(eg. by adding something there if needed) then you can define a public var `LogWIthTraceID` here and the main packages can set it after calling detect.